### PR TITLE
Feature/refactor bdd tests

### PR DESCRIPTION
**Pull request by rhiGriff**
**What is the context of this PR?**

Link to Trello card
https://trello.com/c/eGXufjxE 
Describe what you have changed and why, link to other PRs or Issues as appropriate.
Re ordered the BDD tests to be more readable
**How to review**

Any notes for the reviewer  (include screenshots if appropriate).

### CODE REVIEW

File: tests/behavioural/features/steps/common.py [hunk 1]
```diff
@@ -1,20 +1,56 @@
 from app.application import app
-from behave import given
+from behave import given, when, then
```
Comment (dcdarrell9) [inline tests/behavioural/features/steps/common.py]: Do you need when in here?

File: tests/behavioural/features/steps/draft_get.py [hunk 2]
```diff
@@ -64,26 +64,14 @@ def step_impl_assert_draft_is_retuned(context):
     nose.tools.assert_equal(response['msg_id'], context.resp_data['msg_id'])
```
Comment (dcdarrell9) [inline tests/behavioural/features/steps/draft_get.py]: 'returned' instead of 'retuned' ^^

File: tests/behavioural/features/steps/draft_get.py [hunk 3]
```diff
@@ -64,26 +64,14 @@ def step_impl_assert_draft_is_retuned(context):
...
-@then('a success response is returned')
-def step_impl_assert_200_returned(context):
-    nose.tools.assert_equal(context.response.status_code, 200)
-
-
-#   Scenario: User requests draft that does not exist
-
+# Scenario 2: User requests draft that does not exist
 @given('a user wants a draft that does not exist')
 def step_impl_user_request_non_existant_draft(context):
```
Comment (dcdarrell9) [inline tests/behavioural/features/steps/draft_get.py]: 'existent' ^

File: tests/behavioural/features/steps/draft_get.py [hunk 4]
```diff
@@ -39,7 +39,7 @@ def update_encrypted_jwt():
     database.db.create_all()
```
Comment (dcdarrell9) [inline tests/behavioural/features/steps/draft_get.py]: I think `def update_encrypted_jwt():` is the same in draft_get and draft_post, maybe worth looking at?

File: tests/behavioural/features/steps/draft_post.py [hunk 5]
```diff
@@ -231,28 +251,7 @@ def step_implmsg_id_returned(context):
     nose.tools.assert_true(resp_data['msg_id'] is not None)
```
Comment (dcdarrell9) [inline tests/behavioural/features/steps/draft_post.py]: Should this check if it's actually a valid msg_id?

File: tests/behavioural/features/steps/message_get.py [hunk 6]
```diff
@@ -40,7 +40,66 @@ def update_encrypted_jwt():
...
+    token_data['role'] = 'internal'
+    headers['Authorization'] = update_encrypted_jwt()
+    context.response = app.test_client().get(url.format(context.msg_id), headers=headers)
+
+
+# Scenario 7: Internal user sends message and respondent retrieves the same message with it's labels
+@then("the retrieved message should have the labels INBOX and UNREAD")
+def step_impl_the_retrieved_message_should_havethe_labels_inbox_and_unread(context):
+    response = flask.json.loads(context.response.data)
+    nose.tools.assert_true(len(response['labels']), 2)
```
Comment (dcdarrell9) [inline tests/behavioural/features/steps/message_get.py]: What's the two here for?

File: tests/behavioural/features/steps/drafts_get.py [hunk 7]
```diff
@@ -73,8 +56,7 @@ def step_impl_only_the_users_drafts_are_returned(context):
     nose.tools.assert_equal(len(response['messages']), 7)
```
Comment (dcdarrell9) [inline tests/behavioural/features/steps/drafts_get.py]: What's the 7 for here?

File: tests/behavioural/features/steps/drafts_get.py [hunk 8]
```diff
@@ -92,9 +74,8 @@ def step_impl_user_will_get_drafts_from_second_page_of_pagination(context):
 
     nose.tools.assert_equal(len(response['messages']), 3)
```
Comment (dcdarrell9) [inline tests/behavioural/features/steps/drafts_get.py]: And this 3?

File: tests/behavioural/features/steps/thread_get.py [hunk 9]
```diff
@@ -35,9 +35,47 @@ def update_encrypted_jwt():
...
+def step_impl_assert_all_messages_from_conversation_are_received(context):
+    response = flask.json.loads(context.response.data)
+    nose.tools.assert_equal(len(response), 5)
+
+
+# Scenario 5: Respondent and internal user have a conversation and internal user retrieves that conversation from multiple conversations
+@given("internal user has multiple conversations")
+def step_impl_internal_user_has_multiple_conversations(context):
+
+    for _ in range(0, 2):
```
Comment (emilioward) [inline tests/behavioural/features/steps/thread_get.py]: Why is it `for _ in range` could there be a better variable to use?

Comment (dcdarrell9) [inline tests/behavioural/features/steps/thread_get.py]: Could it just be `for conversation in range(0, 2)` ?

Comment (dcdarrell9) [inline tests/behavioural/features/steps/thread_get.py]: I don't know why `_` is there, but `for x in range` is used other places, keep consistent?

**Pull request merged**